### PR TITLE
ignore blank resource template IDs in profile files when validating

### LINKED ISSUE #1599: Validate that value template refs are not ""
SAG_resourceTemplate_bf2_Serial_Work_json_—___Downloads

results in an error message:
`There was a problem retrieving SAG:resourceTemplate:bf2:Serial:Work: The following referenced resource templates are not available in Sinopia: `
which is not helpful.

### PR DESCRIPTION
**Pull request by peetucket**
fixes #1599

### CODE REVIEW

File: src/ResourceTemplateValidator.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ const validateTemplateRefsExist = async (resourceTemplate) => {
     const refs = propertyTemplate.valueConstraint?.valueTemplateRefs || []
     await Promise.all(refs.map(async (resourceTemplateIdRef) => {
       const found = await foundResourceTemplate(resourceTemplateIdRef)
-      if (!found) {
+      if (!found && resourceTemplateIdRef) { // ignore blank referenced templates elements
```
Comment (justinlittman) [inline src/ResourceTemplateValidator.js]: For clarity, please consider using lodash: `! _.isEmpty(resourceTemplateIdRef)`.

File: src/ResourceTemplateValidator.js [hunk 2]
```diff
@@ -143,11 +143,13 @@ const validateUniqueResourceURIs = async (resourceTemplate) => {
     const refs = propertyTemplate.valueConstraint?.valueTemplateRefs || []
     const resourceURIs = {}
     await Promise.all(refs.map(async (resourceTemplateIdRef) => {
-      const resourceTemplateRef = await fetchResourceTemplate(resourceTemplateIdRef)
-      if (resourceTemplateRef) {
-        const resourceURI = resourceTemplateRef.resourceURI
-        if (!resourceURIs[resourceURI]) resourceURIs[resourceURI] = []
-        resourceURIs[resourceURI].push(resourceTemplateIdRef)
+      if (resourceTemplateIdRef) { // ignore blank referenced templates elements
```
Comment (justinlittman) [inline src/ResourceTemplateValidator.js]: For clarity, please consider adding this as a `filter()` on the refs. For example: `refs.filter(resourceTemplateIdRef => !_.isEmpty(resourceTemplateIdRef).map(...)`.

**Comment by peetucket** [comment]
pushed new commit with reviewer suggestions

**Comment by peetucket** [comment]
> You're nicer than me. I would have failed validation and made the user fix the RT.

Yeah, based on my reading of the original ticket #1599, this situation can come about inadvertently based on what the user does in the profile editor, so I thought it would be nice to just ignore it (since its pretty clear a blank value is of no interest anyway).

**Pull request merged**